### ISSUE TITLE
[Bug] missing AppResources key causes Translate XAML extension to crash the app

### ISSUE THREAD

**Issue by mrbelk** [xamarin/XamarinCommunityToolkit#1813]
### Description

when using the Translate XAML extension and the resource key is missing from the resources file, the extension will throw an exception that causes the app to crash

### Steps to Reproduce

1. set up a Xamarin.Forms project using the LocalizationResourceManager and add some resources to AppResources.resx
2. in the main page's XAML, specify a label with the text being translated using the Translate extension but enter a resource key that is missing from the AppResources.resx file.
3. run the app and watch it crash.

### Expected Behavior
I think it would be good for the extension to show the missing resource key specified, possibly wrapped in asterisks or something to show that the resource key was missing, rather than crashing the entire app

### Actual Behavior
the app crashes

### Basic Information

- Version with issue: 1.3+
- Last known good version: don't know, but probably never
- IDE:VS2019 for Max
- Platform Target Frameworks: 
  - iOS:  15.2
  - Android: 11
  - UWP: N/A 
- Android Support Library Version: 
- Nuget Packages: Xamarin.Forms 5.0.0.2337, Xamarin.CommunityToolkit 2.0.0, SyncFusion.Xamarin 19.4.x.x
- Affected Devices: all

### Workaround
N/A

**Comment by bijington** [comment]
@mrbelk thanks for the report. Out of interest do you get a meaningful exception message as part of the crash?

**Comment by mrbelk** [comment]
Yes, it’s a NullReferenceException that includes the resource key name, but Xamarin Forms, so it’s easy to identify the offender, but obviously it’s after the fact.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by VladislavAntonyuk** [comment]
![image](https://user-images.githubusercontent.com/33021114/154971634-316ff30e-2503-4033-b8de-73227a9ce1eb.png)
NRE is expected here. The error message should contain missed text

**Comment by mrbelk** [comment]
The compile-time checking captures this very well in code; but it's "rude" to throw an exception rendering an entire page because of a missing resource key (when that's not really a "fatal" type of condition that warrants a response like that, IMO). Simply informing the UI in a kinder, gentler way that the resource key is missing, while telling the viewer which key was actually missing, makes more sense to me.

**Comment by pictos** [comment]
What about throwing the exception on Debug and in Release we just return the Key? I used this approach in several of my apps.

**Comment by VladislavAntonyuk** [comment]
Sounds reasonable to me
